x-pack/plugins/alerting/server/routes/rule/apis/find/find_rules_route.ts

```diff
@@ -7,26 +7,40 @@
 
 import { IRouter } from '@kbn/core/server';
 import { UsageCounter } from '@kbn/usage-collection-plugin/server';
+import { ILicenseState } from '../../../../lib';
+import { verifyAccessAndContext } from '../../../lib';
+import { findRulesRequestQuerySchemaV1 } from '../../../../../common/routes/rule/apis/find';
 import type {
   FindRulesRequestQueryV1,
   FindRulesResponseV1,
 } from '../../../../../common/routes/rule/apis/find';
-import { findRulesRequestQuerySchemaV1 } from '../../../../../common/routes/rule/apis/find';
 import { RuleParamsV1, ruleResponseSchemaV1 } from '../../../../../common/routes/rule/response';
-import { ILicenseState } from '../../../../lib';
-import { AlertingRequestHandlerContext, BASE_ALERTING_API_PATH } from '../../../../types';
-import { verifyAccessAndContext } from '../../../lib';
+import {
+  AlertingRequestHandlerContext,
+  BASE_ALERTING_API_PATH,
+  INTERNAL_ALERTING_API_FIND_RULES_PATH,
+} from '../../../../types';
 import { trackLegacyTerminology } from '../../../lib/track_legacy_terminology';
 import { transformFindRulesBodyV1, transformFindRulesResponseV1 } from './transforms';
 
-export const findRulesRoute = (
-  router: IRouter<AlertingRequestHandlerContext>,
-  licenseState: ILicenseState,
-  usageCounter?: UsageCounter
-) => {
+interface BuildFindRulesRouteParams {
+  licenseState: ILicenseState;
+  path: string;
+  router: IRouter<AlertingRequestHandlerContext>;
+  excludeFromPublicApi?: boolean;
+  usageCounter?: UsageCounter;
+}
+
+const buildFindRulesRoute = ({
```

Comment:
These are a lot of changes, can you please update the tests in `x-pack/plugins/alerting/server/routes/rule/apis/find/find_rules_route.test.ts`?